configure.ac

```diff
@@ -34,6 +34,14 @@ dnl Checks for programs
 AC_PROG_CC
 AM_PROG_CC_C_O
 
+AC_ARG_WITH([libfabric],
+            AC_HELP_STRING([--with-libfabric], [Use non-default libfabric location - default NO]),
+            [AS_IF([test -d $withval/lib64], [fab_libdir="lib64"], [fab_libdir="lib"])
+             CPPFLAGS="$CPPFLAGS -I $withval/include"
```

Comment:
You should prefix CPPFLAGS with the -I value, not suffix it.  The use of --with-libfabric is not only to specify where a libfabric install is located when you have no system-installed libfabric, but also to *replace* a system-installed libfabric with an alternate version (by prefixing CPPFLAGS and LDFLAGS).